### ISSUE TITLE
Action does not validate binary is pulled/working as expected

### ISSUE THREAD

**Issue by mjnagel** [defenseunicorns/setup-uds#11]
We encountered an error where Renovate picked up a new `uds-cli` version before the artifact was available in GitHub. To simulate this error you could run the setup-uds action with a version that does not exist yet. The action will complete successfully but you will see an error similar to [this one](https://github.com/defenseunicorns/uds-core/actions/runs/12377801015/job/34548186580?pr=1125#step:3:81) when attempting to run `uds` commands.

This can be seen in the following output which mimics the commands that this action runs:
```console
# Example of the output when curling a non-existent release
❯ curl -sL "https://github.com/defenseunicorns/uds-cli/releases/download/v0.21.0/uds-cli_v0.21.0_Darwin_arm64"
Not Found%    

# Commands the action would run                                                                                                                                                                                                                       
❯ curl -sL "https://github.com/defenseunicorns/uds-cli/releases/download/v0.21.0/uds-cli_v0.21.0_Darwin_arm64" -o uds
❯ chmod +x uds

# Now trying to use UDS CLI
❯ ./uds
./uds: line 1: Not: command not found
```

I believe this curl command 404s so tacking on `--fail` to the curl args would allow it to fail the action. Unsure if this makes the most sense or if the binary should be validated against a sha or something similar which would also catch this.

**Comment by catsby** [comment]
PR to address this in https://github.com/defenseunicorns/setup-uds/pull/12

**Comment by catsby** [comment]
#12 is merged and out in v1.0.1, thanks @mjnagel!
